production.server.project.builder.sh

```diff
@@ -0,0 +1,445 @@
+    printf "\n\e[34;107mWhat branch would you like to use for this production build?\033[0m\n"
+    read branch
+    BRANCH_NAME=$branch
+    printf "\n\e[34;107mAre you absolutely certain that you want to build and release $PROJECT_NAME on this Production Server\033[0m\n\e[34;107mBased on your specified branch of $BRANCH_NAME ?\033[0m"
+    if [[ $PROJECT_USES_WWAPPBASE_SYMLINK = 'yes' ]]; then
+        printf "\n\t\e[34;107mFurther, are you certain that the branch $BRANCH_NAME exists in the wwappbase.js repo?\033[0m\n"
+    fi
+    if [[ $PROJECT_USES_BOB = 'yes' ]]; then
+        printf "\n\t\e[34;107mFurther, are you certain that the branch $BRANCH_NAME exists in the open-code, elasticsearch-java-client, and winterwell-code repos?\033[0m\n"
+    fi
+    printf "\n\nyes/no\n"
+    read answer
+    ANSWER=$answer
+    case $ANSWER in
+        yes|YES)
+            printf "\n\n\t\tContinuing Production Build\n\n"
+        ;;
+        no|NO)
+            printf "\n\n\t\tAborting Operation\n\n"
+            exit 0
+        ;;
+        *)
+            printf "\n\n\t\tAnswer not understood.  Aborting Operation\n\n"
+            exit 0
+        ;;
+    esac
+}
+
+## Checking the immediate logged output for errors or warnings - This Function's Version is 0.02
+function catch_JVM_success_or_error {
+    if [[ $PROJECT_USES_BOB = 'yes' ]]; then
+        INITIAL_LOG_NUM_LINES=$(wc -l $PROJECT_LOG_FILE | awk '{print $1}')
+        while read -t 10 line; do
+            case "$line" in
+                *"AMain Running"* )
+                    printf "\n\t$PROJECT_NAME 's JVM reports a successful startup\n"
+                    exit
+                ;;
+                *"ES.init To reindex"* )
+                    printf "\n\t\e[30;41m$PROJECT_NAME reports that at least one ES index will need to be re-indexed and re-aliased\e[0m\n"
+                    printf "You'll need to read the logged output of the JVM in order to see what exactly needs to be changed\n"
+                    exit
+                ;;
+            esac
+        done < <(tail --lines=+$INITIAL_LOG_NUM_LINES -f $PROJECT_LOG_FILE)
+        RETVAL=$?
+        case $RETVAL in
+            0)
+                echo ""
+            ;;
+            *)
+                printf "The JVM was given 10 seconds to report either success or that an elasticsearch index requires a re-index and re-aliasing before it could start. No such indication was received and parsed.  Please check your service and the log file for this project\n"
+            ;;
+        esac
+    fi
+}
+
+
+
+################
+### Run the Functions in Order
+################
+get_branch_and_print_warning
+check_bob_exists
+check_jerbil_exists
+check_maven_exists
+check_java_home
+check_nodejs_version
+check_for_wwappbasejs_location
+check_for_code_repo_in_bobwarehouse
+cleanup_repo
+cleanup_wwappbasejs_repo
+cleanup_bobwarehouse_repos
+git_checkout_release_branch
+use_bob
+use_npm
+use_webpack
+use_jerbil
+stop_service
```

Comment:
@winterstein , by building/re-building JAR files which are being used by a running JVM, you risk de-stabilizing the service, and causing unknown amounts of fallout.  Anything from incorrectly logged data, dataloss, and yeti bugs can be generated from performing this order of operations in this way.